tests/testsuite.src/used_binaries.at

```diff
@@ -1172,3 +1148,192 @@ AT_CHECK([$COMPILE_MODULE -Wno-unfinished --copy "f.copy" -fstatic-call prog2.co
 )
 
 AT_CLEANUP
+
+
+AT_SETUP([output dependencies])
+
+AT_CAPTURE_FILE([compiler.output])
+# AT_KEYWORDS([])
+AT_DATA([prog.cob], [
+        IDENTIFICATION DIVISION.
+        PROGRAM-ID. prog.
+        DATA DIVISION.
+        WORKING-STORAGE SECTION.
+	COPY COPY1.
+        PROCEDURE DIVISION.
+        MAIN-PROC SECTION.
+        00.
+        COPY COPY2.
+        END-PROC  SECTION.
+        COPY COPY3 in "sub".
+        EX.
+            STOP RUN.
+])
+
+AT_CHECK([mkdir -p sub])
+AT_DATA([COPY1.CPY], [])
+AT_DATA([COPY2.CPY], [       DISPLAY "Hello".
+])
+AT_DATA([sub/COPY3.CPY], [])
```

Comment:
Please include in this test a copy that is included in another copy (=nested)= and include COPY2 two times in the program to cover these edge cases, possibly one time as `COPY COPY2.CPY`.

Also missing: compiling two programs at once. If I understand that correctly, this should result in two dependency files..

Reply:
> Also missing: compiling two programs at once. If I understand that correctly, this should result in two dependency files..

This is actually already tested (using the same COBOL source twice):
```
AT_CHECK([$COMPILE -M prog.cob prog.cob > compiler.output], [0])
```
We do have an output for each .cob file.

Reply:
>  and include COPY2 two times in the program to cover these edge cases

And indeed, we end up with duplicates in the dependency list. I have to fix that.

Reply:
Seems this is because `pp_text_list_add` does not check for duplicates. This function was present before this patch and seems to be only used to build the dependency list: is it okay to change it so that it check for duplicates *unconditionally* ?

Reply:
> > Also missing: compiling two programs at once. If I understand that correctly, this should result in two dependency files..
> 
> This is actually already tested (using the same COBOL source twice):
> 
> ```
> AT_CHECK([$COMPILE -M prog.cob prog.cob > compiler.output], [0])
> ```
> 
> We do have an output for each .cob file.

Hm, not sure if this would be the case for GCC? I've thought about two different programs as this is less questionable and more descriptive.

Reply:
I changed it to have two different programs.